R/transactions.R

```diff
@@ -43,4 +57,29 @@ setGeneric("dbCommit",
 setGeneric("dbRollback",
   def = function(conn, ...) standardGeneric("dbRollback"),
   valueClass = "logical"
-)
\ No newline at end of file
+)
+
+#' @aliases withTransaction,DBIConnection-method
+#' @export
+#' @rdname transactions
+setGeneric("withTransaction",
+  def = function(conn, statement, ...) standardGeneric("withTransaction"),
+  valueClass = "logical"
+)
+
+#' @export
+setMethod("withTransaction", "DBIConnection", function(conn, statement, ...) {
```

Comment:
For single statements, you don't usually need a transaction:

```
withTransaction(con, {
  dbExecQuery(con, "INSERT INTO cars (speed, dist) VALUES (1, 1);"))
  dbExecQuery(con, "INSERT INTO cars (speed, dist) VALUES (2, 2);"))
  dbExecQuery(con, "INSERT INTO cars (speed, dist) VALUES (3, 3);"))
})
```

Requiring a function here means more typing effort, and you won't be able to access the `con` variable.